File: BankApp4-main/ksyama-banks-ui/src/app/core/services/transaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

import { environment } from '../../../environments/environment';
import {
  DepositRequest,
  TransactionResponse,
  TransferRequest,
  WithdrawRequest
} from '../models/transaction.model';

@Injectable({ providedIn: 'root' })
export class TransactionService {
  private readonly url = `${environment.apiBaseUrl}/transactions`;

  constructor(private readonly http: HttpClient) {}

  deposit(request: DepositRequest): Observable<TransactionResponse> {
    return this.http.post<TransactionResponse>(`${this.url}/deposit`, request);
  }

  withdraw(request: WithdrawRequest): Observable<TransactionResponse> {
    return this.http.post<TransactionResponse>(`${this.url}/withdraw`, request);
  }

  transfer(request: TransferRequest): Observable<TransactionResponse> {
    return this.http.post<TransactionResponse>(`${this.url}/transfer`, request);
  }

  getHistory(accountNumber: string): Observable<TransactionResponse[]> {
    return this.http.get<TransactionResponse[]>(`${this.url}/${accountNumber}`);
  }

  getPendingApprovals(): Observable<TransactionResponse[]> {
    return this.http.get<TransactionResponse[]>(`${this.url}/pending`);
  }

  approve(id: number): Observable<TransactionResponse> {
    return this.http.post<TransactionResponse>(`${this.url}/approve/${id}`, {});
  }

  reject(id: number): Observable<TransactionResponse> {
    return this.http.post<TransactionResponse>(`${this.url}/reject/${id}`, {});
  }
}

File: BankApp4-main/bankappv3fin/src/main/java/com/bankapp/api/service/TransactionService.java
package com.bankapp.api.service;

import java.util.List;

import com.bankapp.api.dto.DepositRequest;
import com.bankapp.api.dto.TransactionResponse;
import com.bankapp.api.dto.TransferRequest;
import com.bankapp.api.dto.WithdrawRequest;

public interface TransactionService {

    TransactionResponse deposit(DepositRequest req);

    TransactionResponse withdraw(WithdrawRequest req);

    List<TransactionResponse> history(String accNo);
    
    TransactionResponse transfer(TransferRequest req);

    TransactionResponse approve(Long txnId);

    TransactionResponse reject(Long txnId);
    List<TransactionResponse> getPending();
}